GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/NotificationsViewModel.cs

```diff
@@ -0,0 +1,207 @@
+    {
+        public ICommand AcceptGalaxy { get; set; }
+        public ICommand DeclineGalaxy { get; set; }
+        public ICommand NotifyUser { get; set; }
+        public ICommand ResetNotifications { get; set; }
+        public ICommand ToggleButtonNotification { get; set; }
+        public ICommand ToggleNotifier { get; set; }
+
+        private ObservableCollection<TableUser> _availableUsers;
+        public ObservableCollection<TableUser> AvailableUsers
+        {
+            get { return _availableUsers; }
+            set
+            {
+                _availableUsers = value;
+                OnPropertyRaised("AvailableUsers");
+            }
+        }
+
+        private ClassificationPanelViewModel _classifier;
+        public ClassificationPanelViewModel Classifier
+        {
+            get { return _classifier; }
+            set
+            {
+                _classifier = value;
+                OnPropertyRaised("Classifier");
+            }
+        }
+
+        private TableUser _cooperatingPeer;
+        public TableUser CooperatingPeer
+        {
+            get { return _cooperatingPeer; }
+            set
+            {
+                _cooperatingPeer = value;
+                OnPropertyRaised("CooperatingPeer");
+            }
+        }
+
+        private bool _hideButtonNotification = false;
+        public bool HideButtonNotification
+        {
+            get { return _hideButtonNotification; }
+            set
+            {
+                _hideButtonNotification = value;
+                OnPropertyRaised("HideButtonNotification");
+            }
+        }
+
+        private NotificationStatus _status = NotificationStatus.ClearNotifications;
+        public NotificationStatus Status
+        {
+            get { return _status; }
+            set
+            {
+                _status = value;
+                OnPropertyRaised("Status");
+            }
+        }
+
+        private bool _openNotifier = false;
+        public bool OpenNotifier
+        {
+            get { return _openNotifier; }
+            set
+            {
+                _openNotifier = value;
+                OnPropertyRaised("OpenNotifier");
+            }
+        }
+
+        private string _suggestedAnswer;
+        public string SuggestedAnswer
+        {
+            get { return _suggestedAnswer; }
+            set
+            {
+                _suggestedAnswer = value;
+                OnPropertyRaised("SuggestedAnswer");
+            }
+        }
+
+        private TableUser _user;
+        public TableUser User
+        {
+            get { return _user; }
+            set
+            {
+                _user = value;
+                OnPropertyRaised("User");
+            }
+        }
+
+        public NotificationsViewModel(TableUser user, ObservableCollection<TableUser> allUsers, ClassificationPanelViewModel classifier)
+        {
+            Classifier = classifier;
+            user.Notifications = this;
```

Comment:
Can you go into more detail on why the user this class both need references to each other. 

Also, I think it's a little confusing that the constructor for `NotificationsViewModel` modifies a value of one of the parameters passed into is. Generally, I wouldn't expect a constructor to have this kind of side effect.

Reply:
Yeah, this is a bit iffy I'll admit, and it does create a dependency that might be avoidable. It's tricky with the notifier needing to send a message to the classifier and user. Let me see if I can use the Messenger object to take care of this functionality.